lesson03/assignment/index.js

```diff
@@ -0,0 +1,111 @@
+var arr = []
+
+/*  
+ * Collects data from external JSON file and stores it in an array
+ * 
+ */
+function buildData() {
+    //alert("buildData");
+        d3.json("data/data.json", function (error, data) {
+            if (error)
+                throw error;
+            $.each(data, function (i) {
+
+                arr.push(makePerson(data[i].name, data[i].jobTitle, data[i].Company, data[i].Experience, data[i].School, data[i].Major, data[i].Email, data[i].LinkedInUrl));
+            });
+        });
+}
+
+/*
+ *  Makes a Person object 
+ * 
+ */
+function makePerson(name, jobtitle, company, exp, school, major, email, url){
+    return{
+        name,
+        jobtitle,
+        exp,
+        company,
+        school,
+        major,
+        email,
+        url
+    }
+}
+
+/*
+ * Adds text to the HTML page
+ * 
+ */ 
+function addText() {

```

Comment:
It looks like you were going for an approach of loading up the components and then filling in the data on the fly. We can avoid this by using an array method like `map()` to construct the card data based on that JSON data variable. That way the data is 'imprinted' on the components and we only need to use a single function to build them before we append.

Reply:
What do you mean? Are you looking for the entire card stored as one string, then printed out? I'm not seeing how using .map() would go any faster/consistent...

Reply:
Correct, the goal was to keep the card HTML stored in a single ES6 template literal and use string interpolation to fill out the data. You can do this imperatively without map, but the idea was to print out cards for each object in the data array, with the corresponding data.

Reply:
Updated